.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: Build package on different Os and Python versions
+
+on : [push, pull_request]
+
+jobs:
+
+  build_package:
+    name: Build ${{ github.event.repository.name }} on ${{ matrix.os }} for Python-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        include:
+          - os: ubuntu-latest
+            path: ~/.cache/pip
+          - os: macos-latest
+            path: ~/Library/Caches/pip
+          - os: windows-latest
+            path: ~\AppData\Local\pip\Cache
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python-${{ matrix.python-version }} 
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Activating Python cache
+        uses: actions/cache@v2
+        id: cache_python
+        continue-on-error: true
+        with:
+          path: ${{ matrix.path }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+
+      - name: Install dependencies for ${{ github.event.repository.name }}
+        run: |
+          # The --upgrade and --upgrade-strategy eager flags ensure that pip will always install the latest allowed version of all the dependencies.
+          python -m pip install --upgrade --upgrade-strategy eager -r requirements.txt
+
+      - name: Build ${{ github.event.repository.name }}
+        run: |
+          python -m pip install -e ".[doc, tests]"
+
+      - name: Install dependencies for testing and validation
+        run: |
+          python -m pip install pytest pytest-cov pytest-timeout pytest-benchmark hypothesis
```

Comment:
I think these (apart from `hypothesis`) are installed in the previous step because of
https://github.com/giotto-ai/pyflagser/blob/39f40a48109c3297da7115c36369903221adc238/setup.py#L51-L56
The reason this was done explicitly before is that we were installing the library and its dependencies in a docker image for manylinux, but after leaving the docker image the environment changes (I think) and the dependencies are no longer there.

Btw, we don't need `hypothesis` in `pyflagser` at the moment.

Reply:
From the installation `python -m pip install -e ".[doc, tests]"` because we add the test, it should install the requirements `EXTRAS_REQUIRE`, right ?
I can remove this then

Reply:
Again you were right, this is now removed !